src/store/document-drive.ts

```diff
@@ -137,19 +145,8 @@ export function useFileNodeDocument(
     const documentType =
         kind === 'FILE' ? selectedNode?.documentType : undefined;
 
-    const selectedDocument = useAtomValue(selectedDocumentAtom);
-    const setSelectedDocument = useMemo(() => {
-        function setSelectedDocument(document: Document | undefined) {
-            if (!document) {
-                setFileNodeDocument(undefined);
-            } else if (fileNodeDocument) {
-                setFileNodeDocument({ ...fileNodeDocument, document });
-            } else {
-                throw new Error('fileNodeDocument is undefined');
-            }
-        }
-        return setSelectedDocument;
-    }, [fileNodeDocument, setFileNodeDocument]);
```

Comment:
fileNodeDocument was causing setSelectedDocument to be rebuilt when the document was updated. This change makes it so that strandUpdates is only subbed once per document.